usbsdmux/__main__.py

```diff
@@ -64,17 +64,38 @@ def main():
     ctl = UsbSdMux(args.sg)
     mode = args.mode.lower()
 
-    if mode == "off":
-        ctl.mode_disconnect()
-
-    elif mode in ("dut", "client"):
-        ctl.mode_DUT()
-
-    elif mode == "host":
-        ctl.mode_host()
-
-    elif mode == "get":
-        print(ctl.get_mode())
+    try:
+        if mode == "off":
+            ctl.mode_disconnect()
+
+        elif mode in ("dut", "client"):
+            ctl.mode_DUT()
+
+        elif mode == "host":
+            ctl.mode_host()
+
+        elif mode == "get":
+            print(ctl.get_mode())
+
+    except FileNotFoundError as fnfe:
+        print(fnfe, file=sys.stderr)
+        sys.exit(fnfe.errno)
+    except PermissionError as perr:
+        print(perr, file=sys.stderr)
+        sys.exit(perr.errno)
+    except OSError as ose:
+        if ose.errno == 25:
```

Comment:
I suggest not using the magic number here but instead:

    import errno

    if ose.errno == errno.ENOTTY:
        # An error occured when calling ioctl()
        print(ose, file=sys.stderr)
        …

Reply:
Totally right :) Thx